src/aggregation/intermediate_agg_result.rs

```diff
@@ -77,8 +78,17 @@ impl std::hash::Hash for IntermediateKey {
 
 impl IntermediateAggregationResults {
     /// Add a result
-    pub fn push(&mut self, key: String, value: IntermediateAggregationResult) {
-        self.aggs_res.push(key, value);
+    pub fn push(&mut self, key: String, value: IntermediateAggregationResult) -> crate::Result<()> {
+        let entry = self.aggs_res.entry(key);
+        match entry {
+            Entry::Occupied(mut e) => {
```

Comment:
Can you add a comment explaining when this happens? (I assume this has to do with having an aggregation running on several columns of different types?)

Reply:
This is just an preparation PR for term aggs over different types, it doesn't happen currently